xbmc/cores/VideoPlayer/Process/VideoBuffer.cpp

```diff
@@ -263,9 +266,14 @@ bool CVideoBufferSysMem::Alloc()
     m_image.planesize[2] = 0;
   }
 
-  for (int i = 0; i < 3; ++i)
-    m_image.plane[i] = new uint8_t[m_image.planesize[i]];
+  m_image.plane[0] = m_data;
+  m_image.plane[1] = m_data + m_image.planesize[0];
+  m_image.plane[2] = m_data + m_image.planesize[1];
```

Comment:
This doesn't look right. Perhaps:
m_image.plane[2] = m_image.plane[1] + m_image.planesize[1];